compactor/src/utils.rs

```diff
@@ -63,6 +66,16 @@ impl ParquetFileWithTombstone {
         self.data.id
     }
 
+    /// Return all tombstones in hash map format
+    pub fn tombstones(&self) -> BTreeMap<TombstoneId, Tombstone> {
+        let mut ts_map = BTreeMap::new();
+        for ts in &self.tombstones {
+            ts_map.insert(ts.id, (*ts).clone());
+        }
+
+        ts_map
+    }
```

Comment:
We could store the tombstones as a `BTreeMap` to begin with, if that's more useful! I just picked the simplest data structure based on what I knew at the time :)

If this stays, the code could be simplified as:

```suggestion
    pub fn tombstones(&self) -> BTreeMap<TombstoneId, Tombstone> {
        self.tombstones
            .iter()
            .map(|ts| (ts.id, ts.clone()))
            .collect()
    }
```